util/container/image_test.go

```diff
@@ -0,0 +1,185 @@
+			input:     "k8s.gcr.io/dev/coredns:1.6.2",
+			repo:      "k8s.gcr.io/dev",
+			imageName: "coredns",
+			tag:       "1.6.2",
+			wantError: false,
+		},
+		{
+			name:      "input with name only",
+			input:     "example.com/root",
+			repo:      "example.com",
+			imageName: "root",
+			wantError: false,
+		},
+		{
+			name:      "input with name and tag without path",
+			input:     "example.com/root:tag",
+			repo:      "example.com",
+			imageName: "root",
+			tag:       "tag",
+			wantError: false,
+		},
+		{
+			name:      "input with name and digest without tag",
+			input:     "example.com/root@sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			repo:      "example.com",
+			imageName: "root",
+			digest:    "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			wantError: false,
+		},
+		{
+			name:      "input with path and name",
+			input:     "example.com/user/repo",
+			repo:      "example.com/user",
+			imageName: "repo",
+			wantError: false,
+		},
+		{
+			name:      "input with path, anme and tag",
+			input:     "example.com/user/repo:tag",
+			repo:      "example.com/user",
+			imageName: "repo",
+			tag:       "tag",
+			wantError: false,
+		},
+		{
+			name:      "input with path, name, tag and digest",
+			input:     "example.com/user/repo@sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			repo:      "example.com/user",
+			imageName: "repo",
+			digest:    "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			wantError: false,
+		},
+		{
+			name:      "input with url with port",
+			input:     "url:5000/repo",
+			repo:      "url:5000",
+			imageName: "repo",
+			wantError: false,
+		},
+		{
+			name:      "input with url with port and tag",
+			input:     "url:5000/repo:tag",
+			repo:      "url:5000",
+			imageName: "repo",
+			tag:       "tag",
+			wantError: false,
+		},
+		{
+			name:      "input with url with port and digest",
+			input:     "url:5000/repo@sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			repo:      "url:5000",
+			imageName: "repo",
+			digest:    "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+			wantError: false,
+		},
+		{
+			name:      "input with invalid image name",
+			input:     "url$#",
+			repo:      "",
+			imageName: "",
+			tag:       "",
+			wantError: true,
+		},
+	}
+	for _, tc := range testCases {
+		g := NewWithT(t)
+
+		t.Run(tc.name, func(t *testing.T) {
+			image, err := ImageFromString(tc.input)
+			if tc.wantError {
+				g.Expect(err).To(HaveOccurred())
+			} else {
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(image.Repository).To(Equal(tc.repo))
+				g.Expect(image.Name).To(Equal(tc.imageName))
+				g.Expect(image.Tag).To(Equal(tc.tag))
+				g.Expect(image.Digest).To(Equal(tc.digest))
+				g.Expect(image.String()).To(Equal(tc.input))
+			}
+		})
+	}
+}
+
+func TestModifyImageRepository(t *testing.T) {
```

Comment:
Can we convert this into a table test since we are here? Or should we do that in a separate PR?